src/NameRegistry.sol

```diff
@@ -909,6 +919,39 @@ contract NameRegistry is
         }
     }
 
+    /**
+     * @notice Move the fnames froms the current owners to desired destinations and renew them for another week.
+     *         Does not work when paused because it calls _transfer.
+     *
+     * @param reclaimActions an array of ReclaimAction structs representing the fnames and their corresponding
+     *           destination addresses.
+     */
+    function reclaim(ReclaimAction[] calldata reclaimActions) external payable {
+        // call msg.sender instead of _msgSender() since we don't need meta-tx for admin actions
+        // and it reduces our attack surface area
+        if (!hasRole(MODERATOR_ROLE, msg.sender)) revert NotModerator();
+
+        for (uint256 i = 0; i < reclaimActions.length; i++) {
```

Comment:
consider adding the standard loop gas optimizations - make i unchecked since it is safe, and assign reclaimActions.length outside of the loop. see other for loops in this file for examples

Reply:
sounds good!